logging/log_processing/upload.py

```diff
@@ -79,42 +66,51 @@ def lambda_handler(event, context):
+    json_logging.update_context(
+        aws_request_id=context.aws_request_id,
+        function_name=context.function_name,
+        function_version=context.function_version,
+        log_stream_name=context.log_stream_name,
+    )
+    for i, event_data in enumerate(event["Records"]):
+        logger.info(
+            "Event #{i}: source={eventSource}, name={eventName}, time={eventTime}".format(i=i, **event_data),
+            extra={
+                "event.source": event_data["eventSource"],
+                "event.name": event_data["eventName"],
+                "event.time": event_data["eventTime"],
+            },
+        )
+        bucket_name = event_data["s3"]["bucket"]["name"]
+        object_key = urllib.parse.unquote_plus(event_data["s3"]["object"]["key"])
+        is_log_file = object_key.startswith("_logs/") or "/logs/" in object_key
+        logger.info(f"Bucket={bucket_name}, object={object_key}, is_log_file={is_log_file}")
+        if not is_log_file:
+            continue
 
         file_uri = "s3://{}/{}".format(bucket_name, object_key)
+        processed = compile.load_records([file_uri])
+
         try:
             host, port = config.get_es_endpoint(bucket_name=bucket_name)
             es = config.connect_to_es(host, port, use_auth=True)
             if not config.exists_index_template(es):
                 config.put_index_template(es)
-            processed = compile.load_remote_records(file_uri)
-            ok, errors = index_records(es, processed)
+        except botocore.exceptions.ClientError as exc:
+            logger.warning(f"Failed in initial connection: {exc!s}")
+            continue
+
+        try:
+            index_records(es, processed)
         except parse.NoRecordsFoundError:
-            print("Failed to find records in object '{}'".format(file_uri))
-            return
+            logger.info("Failed to find log records in object '{}'".format(file_uri))
+            continue
         except botocore.exceptions.ClientError as exc:
-            error_code = exc.response['Error']['Code']
-            print("Error code {} for object '{}'".format(error_code, file_uri))
-            return
-
-        body = _build_meta_doc(context, event_data)
-        body["message"] = "Index result for '{}': ok = {}, errors = {}".format(file_uri, ok, errors)
-        body["original_logfile"] = file_uri
-
-        sha1_hash = hashlib.sha1()
-        sha1_hash.update(file_uri.encode())
-        id_ = sha1_hash.hexdigest()
-        res = es.index(index=config.log_index(), body=body, id=id_)
-        print("Sent meta information, result: {}, index: {}".format(res['result'], res['_index']))
-        print("Time remaining (ms):", context.get_remaining_time_in_millis())
+            error_code = exc.response["Error"]["Code"]
+            logger.warning(f"Error code {error_code} for object '{file_uri}'")
+            continue
+
+        logger.info("Indexed log records successfully.", extra={"log_file_uri": file_uri})
```

Comment:
This log-line is in the same loop as the log-line at L76 above, but both lines will not have matching `extra` args.

Reply:
Yes, indeed. There are two problems:
* We only log the information when we have it. For `log_file_uri`, there's much to be done about it unless we want to defer the initial log (which then might be past a line with a bug). For `event.*`, the question is, do we need or want those in the other log lines? How would that help? Do we have other options?
* We could be accidentally inconsistent. Imagine one line having `event.source` and the other `eventSource`. Then discovering the "right" field in a log collection gives a bad experience.

The PR solves the initial intent, turning to JSON-formatting the log lines.

Proposed follow-ups:
* Consider stashing some fields for later log lines. (Pro: search for all log lines tied to a `s3:ObjectPut` event. Con: easy to get wrong and lines that have nothing to do with the event still get shown that way.)
* Consider a schema for the `extra` dict so that there's a published set of additional fields.